docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,342 @@
+---
+sidebar_position: 19
+title: Permissionless ICS
+---
+# ADR 18: Permissionless Interchain Security
+
+## Changelog
+* 27th of June, 2024: Initial draft
+
+## Status
+
+Proposed
+
+## Context
+Currently, a consumer chain can join _Interchain Security_ (ICS) only through a [governance proposal](../features/proposals.md).
+A governance proposal was needed before the introduction of [Partial Set Security](../features/partial-set-security.md) (PSS)
+because validators were required to validate a consumer chain. However, after the introduction of PSS, a consumer chain can
+be either _Top N_ or _Opt In_. If a chain is an Opt In chain, then no validator is required to validate this chain unless they choose to.
+Because of this, we can launch an Opt In consumer chain without going through a governance proposal.
+
+This ADR presents _Permissionless_ ICS, a way in which an [_Opt In_](adr-015-partial-set-security.md) consumer chain can join
+ICS without needing a governance proposal but by simply issuing a transaction.
+
+## Decision
+
+### The Phases of a Consumer Chain
+In Permissionless ICS, launching an Opt In chain is **only** possible through a transaction and not through a [`MsgConsumerAddition`](https://github.com/cosmos/interchain-security/blob/v5.1.0/proto/interchain_security/ccv/provider/v1/tx.proto#L111).
+Nevertheless, ICS does not eliminate the `MsgConsumerAddition` governance proposal, as proposals are still necessary
+for Top N chains. Because of this, this ADR outlines a solution that attempts to preserve as much of the governance proposal code
+as possible.
+Additionally, to make the distinction between governance-proposed versus transaction-launched chains clearer, in Permissionless ICS,
+we can only add, modify, or remove Top N chains with governance proposals (i.e., `MsgConsumerAddition`, `MsgConsumerModification`,
+and `MsgConsumerRemoval`) and we can only add, modify, or remove Opt In chains with transactions.
+Note however that a Top N chain can transform to an Opt In Chain through a `MsgConsumerModification` that sets `top_N == 0` but not vice versa.
+In what follows, what we describe applies mostly for transaction-based (i.e., Opt In) consumer chains, unless stated otherwise.
+
+A consumer chain can reside in four phases: i) waiting, ii) _prelaunched_, iii) _launched_, and iv) _stopped_ phase as seen
```

Comment:
Seems there are extra phases.

I'm seeing these phases:
1. registered
* `consumerId` is assigned,`chain-id` is registered
* only for opt-in
* top_n can skip -> submitting a prop both registers and initializes the chain

2. initialized (pre-launch)
* it's signified by `spawn_time` being populated
* * all top_N start here - they must provide a `spawn_time` anyway
* opt-in chains transition to this stage by executing `MsgInitializeChain`
* spawn_time is present, hash fields are populated and unbonding perios is known

3. launched
* `spawn_time` passed
* IBC clients created
* opted-in validators list has been created and is available through `providerd q provider consumer-genesis`
* chain is live - some validators are running it

Reply:
> Seems there are extra phases.

It seems we still have the same number of phases if you include the _stopped_ phase. Naturally, after a chain has stopped and before having its data fully GCed, it resides in a different phase, what we call in this ADR, the stopped phase.

> top_n can skip -> submitting a prop both registers and initializes the chain

A Top N chain only gets initialized if its gets voted upon, so I think it makes sense to consider it registered before the corresponding proposal passes.

Reply:
Can we have these names for the phases:

- registered (there is a mapping consumerID -> chainID)
- ready-to-launch (there is also a spawn time)
- launched or active or running (there is also a clientID and a validator set)
- stopped

Reply:
I agree with Matija. Rename the phases to registered, initialized, launched, stopped. Also, make consistent throughout the doc.